### PR TITLE
Rs identifiers

### PR DESCRIPTION
**Pull request by rumesh986**
Telegram now requires an admin account which will make the changes to the identifiers file

identifiers has been separated from ROS so it can be used independently for the telegram_admin script

self_id and valid_id are now stored in files and not the launch files
valid_ids can be set from rqt via the admin account so the launch file does not need to be edited directly anymore

### CODE REVIEW

**Review by danipurwadi — APPROVED** [review]

File: launch/ogc_gndtest.launch [hunk 1]
```diff
@@ -10,12 +10,12 @@
...
 	<arg name="identifiers_file" default="$(env HOME)/Sync/identifiers.json" />
-	<arg name="air_ids" default="[4]" />
-	<arg name="self_id" default="1" />
 	<arg name="is_air" default="False" />
+	<arg name="valid_ids_file" default="$(env HOME)/.valid_ids" />
+	<arg name="self_id_file" default="$(env HOME)/.self_id" />
 	
 	<!-- tele args -->
-	<arg name="tdlib_auth_dir" default="$(env HOME)/.tdlib"/>
+	<arg name="tdlib_auth_dir" default="$(env HOME)/.tdlib_gnd"/>
```
Comment (huachen24) [inline launch/ogc_gndtest.launch]: delete _gnd

File: launch/ogc_gndtest.launch [hunk 2]
```diff
@@ -60,11 +60,10 @@
 	<!-- gnd despatcher node -->
 	<node pkg="despatcher" type="gnd_despatcher.py" name="gnd_despatcher" 
 	respawn="true" respawn_delay="3" output="$(arg log_output)">
+		<param name="link_select" value="$(arg link_select)" />
```
Comment (huachen24) [inline launch/ogc_gndtest.launch]: can remove

File: ogc_ws/src/despatcher/src/air_despatcher.py [hunk 3]
```diff
@@ -72,12 +74,19 @@ def __init__(self):
         self.link_select = rospy.get_param("~link_select") # 0 = Tele, 1 = SMS, 2 = SBD
```
Comment (huachen24) [inline ogc_ws/src/despatcher/src/air_despatcher.py]: remove

File: ogc_ws/src/despatcher/src/gnd_despatcher.py [hunk 4]
```diff
@@ -101,10 +102,18 @@ def __init__(self):
         self.pub_to_statustext = rospy.Publisher('ogc/from_despatcher/statustext', String, queue_size=5)
         self.file_to_telegram = rospy.Publisher('ogc/to_telegram/file', LinkMessage, queue_size = 5) # Link to Telegram node
 
-        # Gnd Identifiers
+        # Link switching
+        self.link_select = rospy.get_param("~link_select") # 0 = Tele, 1 = SMS, 2 = SBD
```
Comment (huachen24) [inline ogc_ws/src/despatcher/src/gnd_despatcher.py]: remove

File: ogc_ws/src/identifiers/scripts/identifiers.py [hunk 5]
```diff
@@ -133,12 +140,14 @@ def parse_file(self):
...
 
+		self.admin_id = self.json_obj["admin"]["telegram_id"]
 		self.file_busy = False
 
 	# callback function for the topic subscriber class
 	# used to add contacts to telegram after the telegram node has subscribed to the topic
 	def request_telegram_id(self):
-		rospy.loginfo("Telegram link has subscribed to contact topic")
+		# rospy.loginfo("Telegram link has subscribed to contact topic")
+		print("Telegram link has subscribed to contact topic")
```
Comment (huachen24) [inline ogc_ws/src/identifiers/scripts/identifiers.py]: do we need this?

File: ogc_ws/src/telegram/launch/telegram_air.launch [hunk 6]
```diff
@@ -54,10 +54,8 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 	<!-- air despatcher node -->
 	<node pkg="despatcher" type="air_despatcher.py" name="air_despatcher" respawn="true" respawn_delay="3" output="$(arg log_output)">
-		<rosparam param="ground_ids" subst_value="True">$(arg ground_ids)</rosparam>
 		<param name="interval_1" value="$(arg interval_1)" />
 		<param name="interval_2" value="$(arg interval_2)" />
-		<param name="self_id" value="$(arg self_id)" />
 		<param name="link_select" value="$(arg link_select)" />
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_air.launch]: remove

File: ogc_ws/src/telegram/launch/telegram_air.launch [hunk 7]
```diff
@@ -21,9 +21,10 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
 
-	<arg name="identifiers_file"  default="$(env HOME)/identifiers.json" />
-	<arg name="ground_ids" default="[1]" />
-	<arg name="self_id" default="1" />
+	<arg name="identifiers_file"  default="$(env HOME)/Sync/identifiers.json" />
+	<arg name="valid_ids_file" default="$(env HOME)/.valid_ids" />
+	<arg name="self_id_file" default="$(env HOME)/.self_id" />
+  
 	<arg name="is_air" default="True" />
 	<arg name="link_select" default="0" />
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_air.launch]: remove

File: ogc_ws/src/telegram/launch/telegram_air.launch [hunk 8]
```diff
@@ -21,9 +21,10 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
 	<arg name="waypoint_folder" default="$(env HOME)/Waypoints/"/>
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_air.launch]: Sync

File: ogc_ws/src/telegram/launch/telegram_gnd.launch [hunk 9]
```diff
@@ -17,20 +17,20 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 
 <launch>
-	<arg name="tdlib_auth_dir" default="$(env HOME)/.tdlib"/>
+	<arg name="tdlib_auth_dir" default="$(env HOME)/.tdlib_gnd"/>
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_gnd.launch]: remove _gnd

File: ogc_ws/src/telegram/launch/telegram_gnd.launch [hunk 10]
```diff
@@ -17,20 +17,20 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
 	<arg name="log_output" default="screen"/>
   
 	<arg name="identifiers_file" default="$(env HOME)/Sync/identifiers.json" />
-	<arg name="air_ids" default="[4]" />
-	<arg name="self_id" default="1" />
+	<arg name="valid_ids_file" default="$(env HOME)/.valid_ids" />
+	<arg name="self_id_file" default="$(env HOME)/.self_id" />
+
 	<arg name="is_air" default="False" />
 	<arg name="link_select" default="0" />
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_gnd.launch]: remove

File: ogc_ws/src/telegram/launch/telegram_gnd.launch [hunk 11]
```diff
@@ -40,19 +40,18 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
...
-		<param name="self_id" value="$(arg self_id)" />
-		<rosparam param="valid_ids" subst_value="True">$(arg air_ids)</rosparam>
+		<param name="self_id_file" value="$(arg self_id_file)" />
+		<param name="valid_ids_file" value="$(arg valid_ids_file)" />
 	</node>	
 
 	<!-- ground despatcher node -->
 	<node pkg="despatcher" type="gnd_despatcher.py" name="gnd_despatcher" respawn="true" respawn_delay="3" output="$(arg log_output)">
-		<param name="self_id" value="$(arg self_id)" />
 		<param name="link_select" value="$(arg link_select)" />
```
Comment (huachen24) [inline ogc_ws/src/telegram/launch/telegram_gnd.launch]: remove

**Review by huachen24 — APPROVED** [review]

**Pull request merged**